src/sentry/api/endpoints/project_details.py

```diff
@@ -305,6 +308,9 @@ def put(self, request, project):
                     team_id=old_team_id,
                 ).update(team=new_team)
 
+            if old_slug:
+                project.record_redirect_slug(old_slug)
```

Comment:
nit: I would put this on ProjectRedirect, because otherwise everyone you have a project instance you are bundling this function which is needed vary rarely

Reply:
7a5bb65 agreed.